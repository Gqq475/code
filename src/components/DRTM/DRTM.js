import React from 'react'
import classes from './DRTM.scss'
import DefectJudge from 'components/DefectJudge'
import CriteriaManager from 'components/CriteriaManager'
import CurrentAlarm from 'components/CurrentAlarm'
import AlarmHistory from 'components/AlarmHistory'
import AlarmRate from 'components/AlarmRate'

type Props = {
  params: Object,
};
export class DRTM extends React.Component {
  props: Props

  render () {
    let path = this.props.params.path || 'currentalarm'
    return (
      <div className={classes['DRTM-container']}>
        {
          (() => {
            switch (path) {
              case 'defectjudge':
                return (<DefectJudge data={this.props.drtm}
                  getDefectJudgeById={this.props.getDefectJudgeById}
                  submitJudge={this.props.submitJudge}
                  getDefectDetailById={this.props.getDefectDetailById} />)
              case 'currentalarm':
                return (<CurrentAlarm data={this.props.drtm}
                  getCurrentAlarm={this.props.getCurrentAlarm} />)
              case 'alarmhistory':
                return (<AlarmHistory token={this.props.auth.token} data={this.props.drtm} />)
              case 'criteriamanager':
                return (<CriteriaManager
                  data={this.props.drtm}
                  getCriteriaManagerDetail={this.props.getCriteriaManagerDetail}
                  criteriaManagerUpdate={this.props.criteriaManagerUpdate}
                  criteriaManagerAdd={this.props.criteriaManagerAdd}
                  />)
              case 'alarmrate':
                return (<AlarmRate />)
              default:
            }
          })()
        }
      </div>
    )
  }
}

DRTM.propTypes = {
  auth: React.PropTypes.object.isRequired,
  drtm: React.PropTypes.object.isRequired,
  getCriteriaManagerDetail: React.PropTypes.func.isRequired,
  getDefectJudgeById: React.PropTypes.func.isRequired,
  submitJudge: React.PropTypes.func.isRequired,
  getCurrentAlarm: React.PropTypes.func.isRequired,
  criteriaManagerUpdate: React.PropTypes.func.isRequired,
  criteriaManagerAdd: React.PropTypes.func.isRequired,
  getDefectDetailById: React.PropTypes.func.isRequired
  // getAlarmDetailById: React.PropTypes.func.isRequired
}

export default DRTM
