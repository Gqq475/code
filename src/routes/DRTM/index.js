// import { injectReducer } from '../../store/reducers'
import { TokenIsValid } from 'actions'

export default (store) => ({
  path: 'drtm(/:path)',
  onEnter: function (nextState, replaceState) {
    if (!store.getState().auth.token && !localStorage.getItem('token')) {
      replaceState('/')
    } else {
      store.dispatch(TokenIsValid())
    }
  },
  getComponent (nextState, next) {
    require.ensure([
      './containers/DRTMContainer'
      // './modules/DRTM'
    ], (require) => {
      const DRTM = require('./containers/DRTMContainer').default
      // const reducer = require('./modules/DRTM').default

      // injectReducer(store, { key: 'drtm', reducer })
      next(null, DRTM)
    })
  }
})
