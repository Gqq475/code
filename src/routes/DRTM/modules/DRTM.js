import OurToaster from 'components/OurToaster'
import Utils from '../../../Utils.js'
const qca = __QCA__

export const GET_ALARM_HISTORY_BY_CONDITION = 'DRTM.GET_ALARM_HISTORY_BY_CONDITION'
export const GET_ALARM_DETAIL_BY_ID = 'DRTM.GET_ALARM_DETAIL_BY_ID'
export const DEFECTJUDGE_GETLIST = 'DRTM.DEFECTJUDGE_GETLIST'
export const DEFECTJUDGE_SAVE_TEMPLATE = 'DRTM.DEFECTJUDGE_SAVE_TEMPLATE'
export const CRITERIAMANAGER_GETLIST = 'DRTM.CRITERIAMANAGER_GETLIST'
export const CRITERIAMANAGER_SAVE_TEMPLATE = 'DRTM.CRITERIAMANAGER_SAVE_TEMPLATE'
export const CRITERIAMANAGER_GETDETAIL = 'DRTM.CRITERIAMANAGER_GETDETAIL'
export const CRITERIAMANAGER_GETDETAIL_LOADING = 'DRTM.CRITERIAMANAGER_GETDETAIL_LOADING'
export const CRITERIAMANAGER_UPDATE = 'DRTM.CRITERIAMANAGER_UPDATE'
export const DEFECTJUDGE_GETBYID = 'DRTM.DEFECTJUDGE_GETBYID'
export const CURRENTALARM_GET = 'DRTM.CURRENTALARM_GET'
export const RELOAD_AGGRID = 'DRTM.RELOAD_AGGRID'
export const COLLAPSE_LEFTLAYOUT = 'COLLAPSE_LEFTLAYOUT'
export const GET_DEFECT_DETAIL = 'DRTM.GET_DEFECT_DETAIL'

export function collapseLeftLayout () {
  return (dispatch) => {
    dispatch({
      type: COLLAPSE_LEFTLAYOUT
    })
  }
}

export function getAlarmHistoryByCondition ({ dateStart, dateEnd }) {
  return (dispatch, getState) => {
    let historyApi = `${qca}/alarm/getAlarmList?start_time=${dateStart}&end_time=${dateEnd}`

    fetch(historyApi, {
      headers: {
        'Content-Type': 'application/json',
        'Accept': 'application/json',
        'Authorization': getState().auth.token
      }
    }
    ).then(response => response.json())
    .then((json) => {
      if (json.code === 200000) {
        dispatch({
          type: GET_ALARM_HISTORY_BY_CONDITION,
          data: json.result
        })
      }
    }).catch((error) => {
      console.error(error)
    })
  }
}

export function getDefectJudgeList (state) {
  let urlTemplate = `${qca}` +
    '/glasses?' +
    `start_time=${state.start_time}` +
    `&end_time=${state.end_time}` +
    '&page_no={0}' +
    '&page_size={1}' +
    '&sort_by={2}' +
    '&order={3}'

  state.product && (urlTemplate += `&product=${state.product}`)
  state.operation && (urlTemplate += `&operation=${state.operation}`)
  state.lineID && (urlTemplate += `&line_id=${state.lineID}`)
  state.subEntity && (urlTemplate += `&sub_entity=${state.subEntity}`)

  let url = Utils.format(urlTemplate, [
    state.pageNo,
    state.pageSize,
    state.sortBy,
    state.order
  ])

  return (dispatch, getState) => {
    dispatch({
      type: DEFECTJUDGE_GETBYID,
      data: undefined
    })

    dispatch({
      type: DEFECTJUDGE_SAVE_TEMPLATE,
      data: urlTemplate
    })

    fetch(url, {
      headers: {
        'Content-Type': 'application/json',
        'Accept': 'application/json',
        'Authorization': getState().auth.token
      }
    })
      .then(response => response.json())
      .then((data) => {
        if (data.code === 200000) {
          dispatch({
            type: DEFECTJUDGE_GETLIST,
            data
          })

          dispatch({
            type: RELOAD_AGGRID,
            data: true
          })

          dispatch({
            type: RELOAD_AGGRID,
            data: false
          })
        }
      })
  }
}

export function getDefectJudgeById (id) {
  return (dispatch, getState) => {
    // fetch(`${qca}/k2app-qms/glasses/${id}`)
    fetch(`${qca}/glass?glassId=${id}`, {
      headers: {
        'Content-Type': 'application/json',
        'Accept': 'application/json',
        'Authorization': getState().auth.token
      }
    })
      .then(response => response.json())
      .then((data) => {
        if (data.code === 200000) {
          return dispatch({
            type: DEFECTJUDGE_GETBYID,
            data
          })
        }
      })
  }
}

function getCriteriaManagerListFetch (dispatch, state, getState) {
  let urlTemplate = `${qca}` +
    '/criteria/condition?page_no={0}' +
    '&page_size={1}' +
    '&sort_by={2}' +
    '&order={3}'
  if (state) {
    state.product && (urlTemplate += `&product=${state.product}`)
    state.operation && (urlTemplate += `&operation=${state.operation}`)
    state.lineID && (urlTemplate += `&line_id=${state.lineID}`)
    state.subEntity && (urlTemplate += `&sub_entity=${state.subEntity}`)
    state.dateStart && (urlTemplate += `&start_time=${state.dateStart}`)
    state.dateEnd && (urlTemplate += `&end_time=${state.dateEnd}`)
  }

  dispatch({
    type: CRITERIAMANAGER_SAVE_TEMPLATE,
    data: urlTemplate
  })

  // Must set pageSize as default 1 to get total count
  let url = Utils.format(urlTemplate, [
    state ? state.pageNo : 1,
    state ? state.pageSize : 1,
    state ? state.sortBy : 'id',
    state ? state.order : 'asc'
  ])

  fetch(url, {
    headers: {
      'Content-Type': 'application/json',
      'Accept': 'application/json',
      'Authorization': getState().auth.token
    }
  })
    .then(response => response.json())
    .then(data => {
      if (data.code === 200000) {
        // Utils.updateToken(dispatch, data.result.token)
        dispatch({
          type: CRITERIAMANAGER_GETLIST,
          data
        })

        dispatch({
          type: RELOAD_AGGRID,
          data: true
        })

        dispatch({
          type: RELOAD_AGGRID,
          data: false
        })
      } else {
        OurToaster.show({message: '获取列表失败！'})
      }
    })
}

export function getCriteriaManagerList (state = null) {
  return (dispatch, getState) => {
    getCriteriaManagerListFetch(dispatch, state, getState)
  }
}

export function getCriteriaManagerDetail (id) {
  if (id === null) {
    return {
      type: CRITERIAMANAGER_GETDETAIL,
      data: null
    }
  }
  let url = `${qca}/criteria/id?id=${id}`
  return (dispatch, getState) => {
    dispatch({
      type: CRITERIAMANAGER_GETDETAIL_LOADING,
      data: true
    })
    fetch(url, {
      headers: {
        'Content-Type': 'application/json',
        'Accept': 'application/json',
        'Authorization': getState().auth.token
      }
    })
      .then(response => response.json())
      .then(data => {
        if (data.code === 200000) {
          dispatch({
            type: CRITERIAMANAGER_GETDETAIL_LOADING,
            data: false
          })
          dispatch({
            type: CRITERIAMANAGER_GETDETAIL,
            data
          })
        }
        if (data.code !== 200000) {
          OurToaster.show({message: '获取详情失败！'})
        }
      })
  }
}

export function criteriaManagerUpdate (postData) {
  let url = `${qca}/criteria`
  return (dispatch, getState) => {
    fetch(url, {
      headers: {
        'Content-Type': 'application/json',
        'Accept': 'application/json',
        'Authorization': getState().auth.token
      },
      method: 'PUT',
      body: JSON.stringify(postData)
    })
      .then(response => response.json())
      .then(data => {
        if (data.message === 'Success' && data.result === 1) {
          dispatch({
            type: CRITERIAMANAGER_GETDETAIL,
            data: {result: postData}
          })
          OurToaster.show({message: '更新成功！'})
          dispatch({
            type: RELOAD_AGGRID,
            data: true
          })
          dispatch({
            type: RELOAD_AGGRID,
            data: false
          })
        } else {
          OurToaster.show({message: '更新失败！'})
        }
      })
  }
}

export function criteriaManagerAdd (data) {
  let url = `${qca}/criteria`
  return (dispatch, getState) => {
    fetch(url, {
      headers: {
        'Content-Type': 'application/json',
        'Accept': 'application/json',
        'Authorization': getState().auth.token
      },
      method: 'POST',
      body: JSON.stringify(data)
    })
    .then(response => response.json())
    .then(data => {
      if (data.message === 'Success') {
        OurToaster.show({message: '添加成功！'})
        dispatch({
          type: RELOAD_AGGRID,
          data: true
        })
        dispatch({
          type: RELOAD_AGGRID,
          data: false
        })
      } else {
        OurToaster.show({message: '添加失败!'})
      }
    })
  }
}

export function submitJudge (data, cb) {
  return (dispatch, getState) => {
    fetch(`${qca}/glasses/judge`, {
      method: 'PUT',
      body: JSON.stringify(data),
      headers: {
        'Content-Type': 'application/json',
        'Accept': 'application/json',
        'Authorization': getState().auth.token
      }
    })
      .then(response => response.json())
      .then((data) => {
        cb(data)
      })
  }
}

export function getCurrentAlarm () {
  let url = `${qca}/alarm/getAlarmList?start_time=&end_time=`
  return (dispatch, getState) => {
    fetch(url, {
      headers: {
        'Content-Type': 'application/json',
        'Accept': 'application/json',
        'Authorization': getState().auth.token
      }
    })
      .then(response => response.json())
      .then((data) => {
        return dispatch({
          type: CURRENTALARM_GET,
          data: data.result
        })
      })
  }
}

export function getDefectDetailById (id) {
  return (dispatch, getState) => {
    fetch(`${qca}/defects/byDefectId?param=${id}`, {
      headers: {
        'Content-Type': 'application/json',
        'Accept': 'application/json',
        'Authorization': getState().auth.token
      }
    })
      .then(response => response.json())
      .then((data) => {
        return dispatch({
          type: GET_DEFECT_DETAIL,
          data: data
        })
      })
  }
}

export const actions = {
  getDefectJudgeList,
  getCriteriaManagerList,
  getAlarmHistoryByCondition
  // getAlarmDetailById
}

const ACTION_HANDLERS = {
  [DEFECTJUDGE_GETLIST]: (state, action) => {
    return Object.assign({}, state, {
      glassList: action.data
    })
  },
  [DEFECTJUDGE_SAVE_TEMPLATE]: (state, action) => {
    return Object.assign({}, state, {
      djTemplate: action.data
    })
  },
  [DEFECTJUDGE_GETBYID]: (state, action) => {
    return Object.assign({}, state, {
      defectList: action.data
    })
  },
  [CURRENTALARM_GET]: (state, action) => {
    return Object.assign({}, state, {
      currentAlarmList: action.data
    })
  },
  [CRITERIAMANAGER_GETLIST]: (state, action) => {
    return Object.assign({}, state, {
      criteriaManager: action.data
    })
  },
  [CRITERIAMANAGER_SAVE_TEMPLATE]: (state, action) => {
    return Object.assign({}, state, {
      cmTemplate: action.data
    })
  },
  [CRITERIAMANAGER_GETDETAIL]: (state, action) => {
    return Object.assign({}, state, {
      criteriaManagerDetail: action.data
    })
  },
  [GET_ALARM_HISTORY_BY_CONDITION]: (state, action) => {
    return Object.assign({}, state, {
      alarmHistory: action.data
    })
  },
  [COLLAPSE_LEFTLAYOUT]: (state, action) => {
    return Object.assign({}, state, {
      leftLayoutCollapse: !state.leftLayoutCollapse
    })
  },
  [RELOAD_AGGRID]: (state, action) => {
    return Object.assign({}, state, {
      reloadAGgrid: action.data
    })
  },
  [GET_DEFECT_DETAIL]: (state, action) => {
    return Object.assign({}, state, {
      defectDetail: action.data
    })
  },
  [CRITERIAMANAGER_GETDETAIL_LOADING]: (state, action) => {
    return Object.assign({}, state, {
      createManageGDetail: action.data
    })
  }
}

const initialState = {}
export default function DRTMReducer (state = initialState, action) {
  const handler = ACTION_HANDLERS[action.type]

  return handler ? handler(state, action) : state
}
