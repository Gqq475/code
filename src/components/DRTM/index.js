import DRTM from './DRTM'
export default DRTM
