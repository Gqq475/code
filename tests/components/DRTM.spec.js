import React from 'react'
import { DRTM } from 'components/DRTM/DRTM'

describe('(Component) DRTM', () => {
  it('render normally if path of params is defectjudge', () => {
    const getCriteriaManagerDetail = jest.fn()
    const getDefectJudgeById = jest.fn()
    const submitJudge = jest.fn()
    const getCurrentAlarm = jest.fn()
    const criteriaManagerUpdate = jest.fn()
    const criteriaManagerAdd = jest.fn()
    const getDefectDetailById = jest.fn()
    const data = {
      drtm: {
        a: {
          a: '23'
        },
        b: {
          b: '23'
        }
      },
      params: {
        'path': 'defectjudge'
      },
      auth: {
        'token': 'autodefefdsfsdfdsctjudge'
      }
    }
    const wrapper = shallow(
      <DRTM {...data}
        getCriteriaManagerDetail={getCriteriaManagerDetail}
        getDefectJudgeById={getDefectJudgeById}
        submitJudge={submitJudge}
        getCurrentAlarm={getCurrentAlarm}
        criteriaManagerAdd={criteriaManagerAdd}
        criteriaManagerUpdate={criteriaManagerUpdate}
        getDefectDetailById={getDefectDetailById}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })
  it('render normally if path of params is currentalarm', () => {
    const getCriteriaManagerDetail = jest.fn()
    const getDefectJudgeById = jest.fn()
    const submitJudge = jest.fn()
    const getCurrentAlarm = jest.fn()
    const criteriaManagerUpdate = jest.fn()
    const criteriaManagerAdd = jest.fn()
    const getDefectDetailById = jest.fn()
    const data = {
      drtm: {
        a: {
          a: '23'
        },
        b: {
          b: '23'
        }
      },
      params: {
        'path': 'currentalarm'
      },
      auth: {
        'token': 'autodefefdsfsdfdsctjudge'
      }
    }
    const wrapper = shallow(
      <DRTM {...data}
        getCriteriaManagerDetail={getCriteriaManagerDetail}
        getDefectJudgeById={getDefectJudgeById}
        submitJudge={submitJudge}
        getCurrentAlarm={getCurrentAlarm}
        criteriaManagerAdd={criteriaManagerAdd}
        criteriaManagerUpdate={criteriaManagerUpdate}
        getDefectDetailById={getDefectDetailById}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })
  it('render normally if path of params is alarmhistory', () => {
    const getCriteriaManagerDetail = jest.fn()
    const getDefectJudgeById = jest.fn()
    const submitJudge = jest.fn()
    const getCurrentAlarm = jest.fn()
    const criteriaManagerUpdate = jest.fn()
    const criteriaManagerAdd = jest.fn()
    const getDefectDetailById = jest.fn()
    const data = {
      drtm: {
        a: {
          a: '23'
        },
        b: {
          b: '23'
        }
      },
      params: {
        'path': 'alarmhistory'
      },
      auth: {
        'token': 'autodefefdsfsdfdsctjudge'
      }
    }
    const wrapper = shallow(
      <DRTM {...data}
        getCriteriaManagerDetail={getCriteriaManagerDetail}
        getDefectJudgeById={getDefectJudgeById}
        submitJudge={submitJudge}
        getCurrentAlarm={getCurrentAlarm}
        criteriaManagerAdd={criteriaManagerAdd}
        criteriaManagerUpdate={criteriaManagerUpdate}
        getDefectDetailById={getDefectDetailById}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })
  it('render normally if path of params is criteriamanager', () => {
    const getCriteriaManagerDetail = jest.fn()
    const getDefectJudgeById = jest.fn()
    const submitJudge = jest.fn()
    const getCurrentAlarm = jest.fn()
    const criteriaManagerUpdate = jest.fn()
    const criteriaManagerAdd = jest.fn()
    const getDefectDetailById = jest.fn()
    const data = {
      drtm: {
        a: {
          a: '23'
        },
        b: {
          b: '23'
        }
      },
      params: {
        'path': 'criteriamanager'
      },
      auth: {
        'token': 'autodefefdsfsdfdsctjudge'
      }
    }
    const wrapper = shallow(
      <DRTM {...data}
        getCriteriaManagerDetail={getCriteriaManagerDetail}
        getDefectJudgeById={getDefectJudgeById}
        submitJudge={submitJudge}
        getCurrentAlarm={getCurrentAlarm}
        criteriaManagerAdd={criteriaManagerAdd}
        criteriaManagerUpdate={criteriaManagerUpdate}
        getDefectDetailById={getDefectDetailById}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })
  it('render normally if path of params is alarmrate', () => {
    const getCriteriaManagerDetail = jest.fn()
    const getDefectJudgeById = jest.fn()
    const submitJudge = jest.fn()
    const getCurrentAlarm = jest.fn()
    const criteriaManagerUpdate = jest.fn()
    const criteriaManagerAdd = jest.fn()
    const getDefectDetailById = jest.fn()
    const data = {
      drtm: {
        a: {
          a: '23'
        },
        b: {
          b: '23'
        }
      },
      params: {
        'path': 'alarmrate'
      },
      auth: {
        'token': 'autodefefdsfsdfdsctjudge'
      }
    }
    const wrapper = shallow(
      <DRTM {...data}
        getCriteriaManagerDetail={getCriteriaManagerDetail}
        getDefectJudgeById={getDefectJudgeById}
        submitJudge={submitJudge}
        getCurrentAlarm={getCurrentAlarm}
        criteriaManagerAdd={criteriaManagerAdd}
        criteriaManagerUpdate={criteriaManagerUpdate}
        getDefectDetailById={getDefectDetailById}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
