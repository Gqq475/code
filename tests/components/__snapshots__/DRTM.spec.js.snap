// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) DRTM render normally if path of params is alarmhistory 1`] = `
<div
  className="DRTM-container"
>
  <AlarmHistory
    data={
      Object {
        "a": Object {
          "a": "23",
        },
        "b": Object {
          "b": "23",
        },
      }
    }
    token="autodefefdsfsdfdsctjudge"
  />
</div>
`;

exports[`(Component) DRTM render normally if path of params is alarmrate 1`] = `
<div
  className="DRTM-container"
>
  <Connect(AlarmRate) />
</div>
`;

exports[`(Component) DRTM render normally if path of params is criteriamanager 1`] = `
<div
  className="DRTM-container"
>
  <Connect(CriteriaManager)
    criteriaManagerAdd={[Function]}
    criteriaManagerUpdate={[Function]}
    data={
      Object {
        "a": Object {
          "a": "23",
        },
        "b": Object {
          "b": "23",
        },
      }
    }
    getCriteriaManagerDetail={[Function]}
  />
</div>
`;

exports[`(Component) DRTM render normally if path of params is currentalarm 1`] = `
<div
  className="DRTM-container"
>
  <CurrentAlarm
    data={
      Object {
        "a": Object {
          "a": "23",
        },
        "b": Object {
          "b": "23",
        },
      }
    }
    getCurrentAlarm={[Function]}
  />
</div>
`;

exports[`(Component) DRTM render normally if path of params is defectjudge 1`] = `
<div
  className="DRTM-container"
>
  <Connect(DefectJudge)
    data={
      Object {
        "a": Object {
          "a": "23",
        },
        "b": Object {
          "b": "23",
        },
      }
    }
    getDefectDetailById={[Function]}
    getDefectJudgeById={[Function]}
    submitJudge={[Function]}
  />
</div>
`;
